test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -49,56 +51,137 @@
 (defprotocol Service3
   (service3-fn [this]))
 
-(deftest lifecycle-test
+(defn create-lifecycle-services
+  [call-seq]
+
+  (let [lc-fn (fn [context action] (swap! call-seq conj action) context)]
+    [(service
+       Service1
+       []
+       (init [this context] (lc-fn context :init-service1))
+       (start [this context] (lc-fn context :start-service1))
+       (stop [this context] (lc-fn context :stop-service1))
+       (service1-fn [this] (lc-fn nil :service1-fn)))
+
+     (service
+       Service2
+       [[:Service1 service1-fn]]
+       (init [this context] (lc-fn context :init-service2))
+       (start [this context] (lc-fn context :start-service2))
+       (stop [this context] (lc-fn context :stop-service2))
+       (service2-fn [this] (lc-fn nil :service2-fn)))
+
+     (service
+       Service3
+       [[:Service2 service2-fn]]
+       (init [this context] (lc-fn context :init-service3))
+       (start [this context] (lc-fn context :start-service3))
+       (stop [this context] (lc-fn context :stop-service3))
+       (service3-fn [this] (lc-fn nil :service3-fn)))]))
```

Comment:
I might suggest moving these down so that they're just above the tests that use them.